File: dsa in leetcode/01 - Array LB/p7.cpp
// Rotate Array by One - gfg
// Given an array arr, rotate the array by one position in clockwise direction.

// Examples:

// Input: arr[] = [1, 2, 3, 4, 5]
// Output: [5, 1, 2, 3, 4]
// Explanation: If we rotate arr by one position in clockwise 5 come to the front and remaining those are shifted to the end.
// Input: arr[] = [9, 8, 7, 6, 4, 2, 1, 3]
// Output: [3, 9, 8, 7, 6, 4, 2, 1]
// Explanation: After rotating clock-wise 3 comes in first position.

#include <iostream>
#include<vector>
using namespace std;
 void rotate(vector<int> &arr) {
        // code here
        int ele = arr[arr.size()-1];
        for(int i=arr.size()-1;i>0;i--){
            arr[i] = arr[i-1];
        }
        arr[0] = ele;
    }

int main() {
    
    return 0;
}

File: gfg 160 days/03 - sorting/p7.js
function     mergeArrays(a, b) {
    // code here
    let x = [...a],y=[...b]
    for(let i=y.length-1;i>=0;i--){
        if(y[i]<x[x.length-1]){
            let ele = y[i]
            y[i]=x[x.length-1]
            let k = x.length-2
            while(k>=0 && ele<x[k]){
                x[k+1]=x[k]
                k--
            }
            x[k+1]=ele
        }
    }


    return [x.join(' '),y.join(' ')]

}
function     mergeArrays1(a, b) {
    // code here
    let x = [...a],y=[...b]

    let i=x.length-1,j=0
    while(i>=0 && j<y.length){
        if(x[i]>y[j]){
            [ x[i],y[j] ] = [ y[j],x[i] ]
        }
        i--
        j++
    }
    x.sort((a,b)=>a-b)
    y.sort((a,b)=>a-b)
    return [x.join(' '),y.join(' ')]

}

let a  = [2, 4, 7, 10] , b= [2,3]
// 2 4 7 3
// 3 
// `2 4 7 7
//  2 4 4 7
//  1     5 6 7 8 9
// mergeArrays(a,b)
console.log(mergeArrays(a,b))
console.log(mergeArrays1(a,b))